quiche/src/lib.rs

```diff
@@ -8388,13 +8554,144 @@ impl TransportParams {
                 ),
                 initial_max_streams_bidi: Some(self.initial_max_streams_bidi),
                 initial_max_streams_uni: Some(self.initial_max_streams_uni),
-
-                preferred_address: None,
+                preferred_address,
             },
         )
     }
 }
 
+/// Preferred Address Transport Parameters
+#[derive(Clone, Debug, PartialEq)]
+pub struct PreferredAddressParams {
+    /// The server's preferred V4 address
+    pub addr_v4: Option<SocketAddrV4>,
+    /// The server's preferred V6 address
+    pub addr_v6: Option<SocketAddrV6>,
+    /// The preferred address connection ID
+    pub connection_id: ConnectionId<'static>,
+    /// The preferred address reset token
+    pub stateless_reset_token: u128,
+}
+
+impl PreferredAddressParams {
+    fn new(
+        addr_v4: Option<SocketAddrV4>, addr_v6: Option<SocketAddrV6>,
+        connection_id: ConnectionId<'static>, stateless_reset_token: u128,
+    ) -> Self {
+        PreferredAddressParams {
+            addr_v4,
+            addr_v6,
+            connection_id,
+            stateless_reset_token,
+        }
+    }
+
+    fn decode(b: &mut octets::Octets) -> Result<Self> {
+        let addr_v4 = PreferredAddressParams::decode_v4(b)?;
+        let addr_v6 = PreferredAddressParams::decode_v6(b)?;
+
+        if addr_v4.is_none() && addr_v6.is_none() {
+            return Err(Error::InvalidTransportParam);
+        }
+
+        let cid = b.get_bytes_with_varint_length()?.to_vec().into();
+        let stateless_reset_token = u128::from_be_bytes(
+            b.get_bytes(16)?.to_vec().try_into().unwrap(), /* Since a 128 bit
+                                                            * vector has just
+                                                            * successfully
+                                                            * been created
+                                                            * above, this
+                                                            * should never
+                                                            * panic. */
```

Comment:
Please convert into a comment on the line above

Reply:
Done 👍